rust/src/rfb/rfb.rs

```diff
@@ -227,15 +240,26 @@ impl RFBState {
                             match chosen_security_type {
                                 2 => self.state = parser::RFBGlobalState::TCVncChallenge,
                                 1 => self.state = parser::RFBGlobalState::TSClientInit,
-                                _ => return AppLayerResult::err(),
+                                _ => {
+                                    if let Some(current_transaction) = self.get_current_tx() {
+                                        RFBState::set_event(
+                                            current_transaction,
+                                            RFBEvent::UnimplementedSecurityType,
+                                        );
+                                    }
+                                    // We have just have seen a security type we don't know about.
+                                    // This is not bad per se, it might just mean this is a
+                                    // proprietary one not in the spec.
+                                    // Continue the flow but stop trying to map the protocol.
+                                    self.state = parser::RFBGlobalState::Skip;
```

Comment:
See What SSH does for instance when it gets into encrypted traffic

Reply:
Well, the traffic is not encrypted... :)
I actually did that and my previous approach went into bypass and skip inspection here, but maybe someone wants to do just `content` based pattern matching on RFB traffic, and I do not want to break that. So I deliberately did not do that here.

Reply:
You are right :-)